eva-server/src/main/java/uk/ac/ebi/eva/server/ws/GeneWSServer.java

```diff
@@ -76,19 +73,14 @@ public QueryResponse getVariantsByGene(@PathVariable("geneIds") List<String> gen
 
         MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species));
 
-        List alternates = null;
-        if (alternate != null && !alternate.isEmpty()) {
-            alternates = Collections.singletonList(alternate);
-        }
-
         List<VariantEntityRepositoryFilter> filters =
                 new FilterBuilder().getVariantEntityRepositoryFilters(maf, polyphenScore, siftScore, studies,
-                                                                      consequenceType, reference,
-                                                                      alternates, geneIds);
+                                                                      consequenceType);
 
         List<VariantEntity> variantEntities =
-                variantEntityRepository.findByComplexFilters(filters, Utils.getPageRequest(queryOptions));
-        Long numTotalResults = variantEntityRepository.countByComplexFilters(filters);
+                variantEntityRepository.findByGenesAndComplexFilters(geneIds, filters,
+                                                                     Utils.getPageRequest(queryOptions));
+        Long numTotalResults = variantEntityRepository.countByGenesAndComplexFilters(null, filters);
```

Comment:
Now that the gene IDs are not part of the filters, I think this call needs to be 

```variantEntityRepository.countByGenesAndComplexFilters(geneIds, filters);```